### PR TITLE
feat: add cost report pages, components

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
스크린샷 2024-01-23 오후 7 17 20
스크린샷 2024-01-23 오후 7 17 26

### CODE REVIEW

**Review by skdud4659 — APPROVED** [review]
LGTM~!

File: apps/web/src/common/components/custom-date-modal/CustomDateModal.vue [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+const emit = defineEmits<{(event: 'update:visible', visible: boolean): void;
+    (event: 'confirm', start: string, end: string): void;
+    (event: 'cancel'): void;
+}>();
+
+const state = reactive({
+    proxyVisible: computed({
+        get() { return props.visible; },
+        set(val) { emit('update:visible', val); },
+    }),
```
Comment (piggggggggy) [inline apps/web/src/common/components/custom-date-modal/CustomDateModal.vue]: 🤔

File: apps/web/src/common/components/custom-date-modal/CustomDateModal.vue [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                    centered
+                    size="sm"
+                    fade
+                    backdrop
+                    :visible="state.proxyVisible"
+                    :disabled="state.invalid"
+                    @confirm="handleConfirm"
+                    @update:visible="handleUpdateVisible"
+                    @close="$emit('cancel')"
+                    @cancel="$emit('cancel')"
```
Comment (piggggggggy) [inline apps/web/src/common/components/custom-date-modal/CustomDateModal.vue:114]: Suggestion: how about using function named handlexxxx~?

File: apps/web/src/services/cost-explorer/components/CostReportMonthlyTotalAmountSummaryCard.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    >
+                        {{ $t('BILLING.COST_MANAGEMENT.COST_REPORT.SEE_DETAILS') }}
+                    </p-link>
+                </div>
+                <div class="col-span-12 lg:col-span-6" />
+            </div>
+            <div ref="chartContext"
+                 class="chart"
+            />
+            차아트
```
Comment (piggggggggy) [inline apps/web/src/services/cost-explorer/components/CostReportMonthlyTotalAmountSummaryCard.vue:87]: 🥶🥶🥶🥶

File: apps/web/src/services/cost-explorer/components/CostReportRecipientsModal.vue [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    enableAdditionalRecipients: false,
+    selectedAdminRole: [] as string[],
+    selectedWorkspaceOwner: [] as string[],
+    additionalRecipients: [] as string[],
+});
+
+/* Util */
+const getInvalidText = (value?: string): string => {
+    if (!value) return '';
+    console.log(value, emailValidator(value));
```
Comment (piggggggggy) [inline apps/web/src/services/cost-explorer/components/CostReportRecipientsModal.vue:95]: 🤔

**Review by piggggggggy — APPROVED** [review]
LGTM! I left some comments~!

**Pull request merged**